sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/issue/internal/DefaultIssue.java

```diff
@@ -125,40 +100,45 @@ public RuleKey ruleKey() {
     return this.ruleKey;
   }
 
-  @CheckForNull
   @Override
-  public InputPath inputPath() {
-    return this.path;
+  public Severity overriddenSeverity() {
+    return this.overriddenSeverity;
   }
 
   @Override
-  public Integer line() {
-    return this.line;
+  public Double effortToFix() {
+    return this.effortToFix;
   }
 
-  @Override
-  public String message() {
-    return this.message;
+  public String key() {
+    return this.key;
   }
 
   @Override
-  public Severity overridenSeverity() {
-    return this.overridenSeverity;
+  public List<IssueLocation> locations() {
+    return ImmutableList.copyOf(this.locations);
   }
 
   @Override
-  public Double effortToFix() {
-    return this.effortToFix;
-  }
-
-  public String key() {
-    return this.key;
+  public List<ExecutionFlow> executionFlows() {
+    return Lists.transform(this.executionFlows, new Function<List<IssueLocation>, ExecutionFlow>() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a named "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)